lib/appium_lib/common/element.rb

```diff
@@ -0,0 +1,76 @@
+        @full_source = ::REXML::Document.new @page_source
+        @elements = all
+      end
+
+      private def all
+        ::REXML::XPath.match(@full_source, '//*').select do |node|
+          next if node.is_a? REXML::XMLDecl
+          next unless node.has_attributes?
+
+          node.attributes.size > 1
+        end
+      end
+
+      EVAL_ATTRIBUTES = %w(checkable clickable focusable scrollable long_clickable enabled).freeze
+      def filter_actionable(filter_attributes = EVAL_ATTRIBUTES)
+        @elements = @elements.select { |node|
+          filter_attributes.find { |attribute| node.attributes[attribute] == 'true' }
+        }.select { |node|
+          !node.attributes['bounds'].nil?
+        }
+      end
+
+      ANDROID_EXCLUDE_IDS = %w(android:id/content android:id/navigationBarBackground android:id/parentPanel android:id/topPanel android:id/title_template android:id/contentPanel android:id/scrollView android:id/buttonPanel).freeze
```

Comment:
I was thinking about this. Perhaps we give users an option to include or exclude these locators by passing in an array of excludable locators and not hard code in the class itself. What do you think?

Reply:
Yeah, I agree with it.
The hard code thing is the default behaviour, but it should be customisable.